pkg/controllers/management/k3sbasedupgrade/k3sbased_upgrade_handler.go

```diff
@@ -210,33 +213,18 @@ func (h *handler) deployK3sBasedUpgradeController(clusterName, updateVersion str
 			}
 		}
 
-		desiredApp := app.DeepCopy()
 		externalIDIsCorrect := app.Spec.ExternalID == latestVersionID
-		var pspValuesHaveBeenSet bool
-
-		if is125OrAbove {
-			pspValuesHaveBeenSet = desiredApp.Spec.ValuesYaml == fmt.Sprintf(ConditionalPSPValues, "false")
-		} else {
-			pspValuesHaveBeenSet = desiredApp.Spec.ValuesYaml == fmt.Sprintf(ConditionalPSPValues, "true")
-		}
+		pspValuesHaveBeenSet := app.Spec.ValuesYaml == expectedChartValues
 
 		// everything is up-to-date and PSP attributes are set up properly, no need to update.
 		if externalIDIsCorrect && pspValuesHaveBeenSet {
 			return nil
 		}
 
-		// we only want to disable PSPs for clusters >= 1.25. The default is enabled,
-		// so if we are sub 1.25 we don't need to define a value at all.
-		if !pspValuesHaveBeenSet {
-			if is125OrAbove {
-				desiredApp.Spec.ValuesYaml = fmt.Sprintf(ConditionalPSPValues, "false")
-			} else {
-				desiredApp.Spec.ValuesYaml = fmt.Sprintf(ConditionalPSPValues, "true")
-			}
-		}
-
+		desiredApp := app.DeepCopy()
```

Comment:
Ok these changes in general look better, I was having the feeling the previous commit got a little wordy haha. Looks pretty good

Reply:
Oh sorry, this should be a single commit. I'll fix that